src/qt/walletcontroller.cpp

```diff
@@ -343,9 +344,10 @@ void OpenWalletActivity::open(const std::string& path)
         tr("Opening Wallet <b>%1</b>…").arg(name.toHtmlEscaped()));
 
     QTimer::singleShot(0, worker(), [this, path] {
-        std::unique_ptr<interfaces::Wallet> wallet = node().walletLoader().loadWallet(path, m_error_message, m_warning_message);
+        auto wallet{node().walletLoader().loadWallet(path, m_warning_message)};
 
-        if (wallet) m_wallet_model = m_wallet_controller->getOrCreateWallet(std::move(wallet));
+        if (wallet) m_wallet_model = m_wallet_controller->getOrCreateWallet(wallet.ReleaseObj());
+        else m_error_message = wallet.GetError();
```

Comment:
Below I used `        m_error_message = wallet ? bilingual_str{} : wallet.GetError();`

It would be good to use the same code consistently everywhere.

Reply:
I think the code change was not correctly pushed.
The commit 7cbfa1335bd89d5831e916672e57aa8601b12bb4 still displays the line:

```
else m_error_message = wallet.GetError();
```
and not:
```
m_error_message = wallet ? bilingual_str{} : wallet.GetError();
```

Reply:
@shaavan I said the suggestion was addressed because I understand it's about keeping all code consistent across all commits.

I had originally defined all  `m_error_message` same way as done in https://github.com/bitcoin/bitcoin/pull/25594: ` m_error_message = wallet ? bilingual_str{} : wallet.GetError();`.

So https://github.com/bitcoin/bitcoin/pull/25616#discussion_r921692400 and https://github.com/bitcoin/bitcoin/pull/25616#discussion_r921688995 suggested changing them to `else m_error_message = res_wallet.GetError();` but I had forgotten to change it to `restoreWallet()`. https://github.com/bitcoin/bitcoin/commit/7cbfa1335bd89d5831e916672e57aa8601b12bb4 fixed this.

If I understand the `BResult` interface correctly, it seems to be the clearest option, as [`BResult<T>::GetError()`](https://github.com/bitcoin/bitcoin/blob/826fae6a0fe184518af625ebb6d1ef9e64631385/src/util/result.h#L41-L44) will return the default `bilingual_str{} :` value [if `T` is not set](https://github.com/bitcoin/bitcoin/blob/826fae6a0fe184518af625ebb6d1ef9e64631385/src/util/result.h#L22).

However, both approaches work. If the reviewers have a strong opinion about one of them, I can change it.

Reply:
`BResult` is essentially an std::variant wrapper. It stores one object OR the other (succeed or failure). Not both.

So, it is redundant to ask `m_error_message = wallet ? bilingual_str{} : wallet.GetError();.` first and then ask if succeeded in another if block.

Reply:
`m_error_message = wallet ? bilingual_str{} : wallet.GetError();` was intentional to reset the member that stores the error, instead of leaving the previous value, if no error occurred. However, I am fine changing it to something else.

Reply:
Yeah, I double checked it before comment. The error message is only set here after calling the backend method. So, the string will always be empty if no error occurs.

Reply:
Is it possible to load a wallet with an error and then load a wallet without error on the same walletcontroller?

Reply:
Oh, I missed the comment, sorry Marko.

------

> Is it possible to load a wallet with an error and then load a wallet without error on the same walletcontroller?

You mean, restoring or opening a wallet?

Because if that is the case, there shouldn't be any problem. Every time that we restore or open a wallet we create a new `WalletControllerActivity` subclass which encapsulates the error and warning messages (it's a single shot class. It gets deleted as soon as it finishes processing the action).